### PR TITLE
feat: validate and transform post placement adoption part 3 (ISDB-284)

### PR DESCRIPTION
**Pull request by rahulsanjava19**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ ] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/bulkscan/services/BulkScanA58Service.java [hunk 1]
```diff
@@ -47,6 +47,11 @@ public class BulkScanA58Service implements BulkScanService {
     public static final String STEP_PARENT_ADOPTION = "Step Parent";
 
     public static final String SCAN_DOCUMENTS = "scannedDocuments";
+    public static final String OTHER_PARENT_RELATIONSHIP_TO_CHILD = "otherParentRelationshipToChild";
+    public static final String RELATIONSHIP_FATHER = "child_relationship_father";
+    public static final String RELATIONSHIP_OTHER = "child_relationship_other";
+    public static final String FATHER = "Father";
+    public static final String OTHER_PARENT = "Other parent";
```
Comment (hmcts-kodatala) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/services/BulkScanA58Service.java:54]: pls merge from master A58 services

File: src/test/java/uk/gov/hmcts/reform/bulkscan/utils/TestDataUtil.java [hunk 2]
```diff
@@ -858,4 +858,66 @@ public static List<OcrDataField> getA58Data() {
...
+        birthMotherOccupationField.setName("mother_occupation");
+        birthMotherOccupationField.setValue("testBirthMotherOccupation");
+        fieldList.add(birthMotherOccupationField);
+
+        OcrDataField childRelationshipFatherField = new OcrDataField();
+        childRelationshipFatherField.setName("child_relationship_father");
+        childRelationshipFatherField.setValue("true");
+        fieldList.add(childRelationshipFatherField);
+
+        return fieldList;
```
Comment (hmcts-kodatala) [inline src/test/java/uk/gov/hmcts/reform/bulkscan/utils/TestDataUtil.java:914]: can we use request json for per test case ?

**Review by hmcts-kodatala — COMMENTED** [review]
Pls check review comments

File: src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/controllers/BulkScanEndpointTest.java [hunk 3]
```diff
@@ -9,6 +9,7 @@
 import org.apache.commons.lang3.StringUtils;
 import org.junit.Before;
 import org.junit.Test;
+import org.junit.jupiter.api.DisplayName;
```
[the remark this answers is not in the document]
Comment (rahulsanjava19) [reply src/functionalTest/java/uk/gov/hmcts/reform/bulkscan/controllers/BulkScanEndpointTest.java]: done

File: src/test/java/uk/gov/hmcts/reform/bulkscan/utils/TestDataUtil.java (hunk 2 above)
Comment (rahulsanjava19) [reply src/test/java/uk/gov/hmcts/reform/bulkscan/utils/TestDataUtil.java:914]: done

File: src/main/java/uk/gov/hmcts/reform/bulkscan/services/BulkScanA58Service.java [hunk 4]
```diff
@@ -124,20 +170,86 @@ public BulkScanTransformationResponse transform(BulkScanTransformationRequest bu
...
+        if (TRUE.equalsIgnoreCase(inputFieldsMap.get(applicant_relationToChild_father_partner))) {
+            return RelationToChildEnum.FATHER.getName();
+        }
+        if (TRUE.equalsIgnoreCase(inputFieldsMap.get(applicant_relationToChild_mother_partner))) {
+            return RelationToChildEnum.MOTHER.getName();
+        }
+        if (TRUE.equalsIgnoreCase(inputFieldsMap.get(applicant_relationToChild_non_civil_partner))) {
+            return RelationToChildEnum.CIVIL.getName();
+        }
+        return "";
```
Comment (hmcts-kodatala) [inline src/main/java/uk/gov/hmcts/reform/bulkscan/services/BulkScanA58Service.java:226]: move "" string constants which reuse or we can find string utils have empty constants

**Review by hmcts-kodatala — COMMENTED** [review]
pls check one comments

**Review by hmcts-kodatala — APPROVED** [review]
Ok

**Review by parvezhmcts — APPROVED** [review]
Approved

**Review by saravanakumar-palanichamy — APPROVED** [review]

**Pull request merged**